File: python/rps.py
import random

while True:
    plr_score = 0
    tries = 0
    while tries < 3:
        print("Round", tries + 1)
        plr = input("Rock, Paper, or Scissors? ").lower()
        if plr not in ["rock", "paper", "scissors"]:
            print("Invalid input. Try again.")
            continue
        comp = random.choice(["rock", "paper", "scissors"])
        print("Computer chose", comp)
        if plr == comp:
            print("Tie!")
        elif (plr == "rock" and comp == "scissors") or (plr == "paper" and comp == "rock") or (plr == "scissors" and comp == "paper"):
            print("You win!")
            plr_score += 1
        else:
            print("You lose!")
        tries += 1
    if plr_score >= 2:
        print("=> You win the game! <=")
    else:
        print("=> You lose the game! <=")
    play_again = input("Do you want to play again? (yes/no) ").lower()
    if play_again != "yes":
        break

File: apcs/RPS.java
import java.util.Scanner;
class RPS {
    public static void main(String[] args) {
        String[] options = {"Rock", "Paper", "Scissors"};
        Scanner s = new Scanner(System.in);
        boolean gameActive = true;
        while (gameActive) {
            int currTry = 0;
            int plrWins = 0;
            while (currTry < 3) {
                System.out.println("=> Choose Rock, Paper or Scissors <=");
                String choice = s.nextLine();
                String computerChoice = options[(int) (Math.random() * options.length)];
                System.out.println("Computer chose: " + computerChoice);
                if (choice.equals(computerChoice)) {
                    System.out.println("Draw");
                } else if (choice.equals("Rock") && computerChoice.equals("Scissors") || choice.equals("Paper") && computerChoice.equals("Rock") || choice.equals("Scissors") && computerChoice.equals("Paper")) {
                    plrWins++;
                    currTry++;
                    System.out.println("You won");
                } else {
                    System.out.println("You lost");
                    currTry++;
                }
            }
            System.out.println("=> Game over <=");
            if (plrWins >= 2) {
                System.out.println("$$ You won the game");
            } else {
                System.out.println("%% You lost the game");
            }
            System.out.println("Do you want to play again? (yes/no)");
            String response = s.nextLine();
            if (response.equals("yes")) {
                continue;
            } else {
                gameActive = false;
            }
        }
    }
}